src/main/java/ch/njol/skript/effects/EffHealth.java

```diff
@@ -34,42 +17,46 @@
 import ch.njol.util.Math2;
 import org.bukkit.entity.Damageable;
 import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
 import org.bukkit.inventory.ItemStack;
-import org.eclipse.jdt.annotation.Nullable;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
 
 @Name("Damage/Heal/Repair")
-@Description("Damage/Heal/Repair an entity, or item.")
+@Description("Damage, heal, or repair an entity or item.")
 @Examples({
 	"damage player by 5 hearts",
+	"damage player by 3 hearts with fake cause fall",
 	"heal the player",
 	"repair tool of player"
 })
 @Since("1.0")
 public class EffHealth extends Effect {
-
 	static {
 		Skript.registerEffect(EffHealth.class,
-			"damage %livingentities/itemtypes/slots% by %number% [heart[s]] [with fake cause %-damagecause%]",
+			"damage %livingentities/itemtypes/slots% by %number% [heart[s]] [with [fake] [damage] cause %-damagecause%]",
 			"heal %livingentities% [by %-number% [heart[s]]]",
 			"repair %itemtypes/slots% [by %-number%]");
 	}
 
+	private final boolean canSetDamageCause = Skript.classExists("org.bukkit.damage.DamageSource");
 	private Expression<?> damageables;
-	@Nullable
-	private Expression<Number> amount;
+	private @UnknownNullability Expression<Number> amount;
 	private boolean isHealing, isRepairing;
+	private @UnknownNullability Expression<DamageCause> exprCause = null;
 
 	@SuppressWarnings("unchecked")
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		if (matchedPattern == 0 && exprs[2] != null)
-			Skript.warning("The fake damage cause extension of this effect has no functionality, " +
-				"and will be removed in the future");
+		if (matchedPattern == 0 && exprs[2] != null && !canSetDamageCause)
+			Skript.warning("Using the fake cause extension in effect 'damage' requires Spigot 1.20.4+");
```

Comment:
Do we want to throw an error or the warning? I don't think I'd like code running if there's invalid usage going through. The warning has existed long enough prior that there should be near zero percent concern of breaking change and this would hopefully force version checking or similar for people wanting to support older versions of mc.

Reply:
Yeah I think erroring out here is better. Warning works, but it might be misleading to some users, and it'd make the logic later on easier too.

Reply:
I don't think we should error the code. We also can use `damage` effect even if it doesn't support for `damage cause`